src/backend/distributed/executor/multi_utility.c

```diff
@@ -170,11 +170,19 @@ multi_ProcessUtility(Node *parsetree,
 					 DestReceiver *dest,
 					 char *completionTag)
 {
-	bool schemaNode = SchemaNode();
 	bool commandMustRunAsOwner = false;
 	Oid savedUserId = InvalidOid;
 	int savedSecurityContext = 0;
 
+	if (IsA(parsetree, TransactionStmt))
```

Comment:
Maybe add a comment like
```
/*
 * Transaction statements are run in aborted transactions and such - a lot of checks cannot be done safely in that
 * state. Since we never need to intercept transaction statements, short-circuit.
 */
```
instead of the contained one?